### ISSUE TITLE
URL() constructor

### ISSUE THREAD

**Issue by onlywei** [zloirock/core-js#117]
Hello,

Are there any plans to add the URL constructor to this library?
https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

Thanks.

**Comment by zloirock** [comment]
Thanks for proposal, I will think about it.

**Comment by zloirock** [comment]
I think, it's possible to add minimalistic implementation like that: https://gist.github.com/Yaffle/1088850

**Comment by zloirock** [comment]
In #288 proposed an option.

**Comment by zloirock** [comment]
Now it's an ECMAScript proposal, https://github.com/jasnell/proposal-url.

**Comment by kevipapo1** [comment]
jsdom implemented the URL class to the WHATWG standard: https://github.com/jsdom/whatwg-url

Would you be OK with using their code and formatting it to match the structure of your project? They also use the MIT license.

**Comment by kof** [comment]
what about using something like this as a polyfill? 

```js
import pick from 'lodash/object/pick'
import mapValues from 'lodash/object/mapValues'

const parser = document.createElement('a')

const components = [
  'hash', 'host', 'hostname', 'href', 'pathname', 'port', 'protocol', 'search'
]

export default function parse(url) {
  parser.href = encodeURI(url)
  return mapValues(pick(parser, components), decodeURIComponent)
}
```

**Comment by zloirock** [comment]
@kof at least, `document.createElement('a')` is not available in non-browser engines.

**Comment by zloirock** [comment]
@KevinSnyderCodes feel free to add it after merging `v3` branch with `master`.

**Comment by yvele** [comment]
Is this issue also covering `URLSearchParams` [polyfill](https://github.com/lifaon74/url-polyfill)?

**Comment by zloirock** [comment]
@yvele yes, includes.

**Comment by zloirock** [comment]
Most likely, it will be added soon. I'm working on it.

**Comment by loganfsmyth** [comment]
@zloirock Would this be on by default, or opt in? I wouldn't want users of `babel-polyfill` to load this I don't think, at least not without opting in.

**Comment by zloirock** [comment]
@loganfsmyth it will be available in namespaces `core-js/web` and `core-js/stage/0`.

**Comment by zloirock** [comment]
Available in `core-js@3.0.0-beta.5`.
